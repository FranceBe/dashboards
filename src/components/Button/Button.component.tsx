// Button component
import { ButtonDefault } from 'components/Button/button.style'
import React, { ButtonHTMLAttributes } from 'react'

export const Button: React.FC<ButtonHTMLAttributes<HTMLButtonElement>> = (props) => {
  return <ButtonDefault {...props}>{props.children}</ButtonDefault>
}

export default Button
