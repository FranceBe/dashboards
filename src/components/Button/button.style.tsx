import styled from 'styled-components'
import { elements, palette, spaces } from 'styles/variables'

export const ButtonDefault = styled.button`
  background-color: ${palette.blue_primary};
  color: ${palette.white};
  padding: ${spaces.regular};
  border-radius: ${spaces.small};
  box-shadow: ${elements.box_shadow};
  font-weight: bold;
  cursor: pointer;
  font-size: 1rem;
  &:hover {
    background-color: ${palette.blue_secondary};
  }
  &:disabled,
  &:disabled:hover {
    background-color: ${palette.grey_primary};
    color: grey;
    cursor: not-allowed;
  }
`
