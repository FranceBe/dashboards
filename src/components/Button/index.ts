import ButtonContainer, { Button } from 'components/Button/Button.component'

export { Button }

export default ButtonContainer
