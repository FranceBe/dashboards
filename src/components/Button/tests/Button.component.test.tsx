import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Button from 'components/Button'
import React from 'react'

describe('Button', () => {
  it('should match snapshot when enabled', () => {
    const { container } = render(<Button>Button text</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should match snapshot when disabled', () => {
    const { container } = render(<Button disabled>Button text</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should call provided function on click', () => {
    const mockedFn = jest.fn()
    render(<Button onClick={mockedFn}>Button text</Button>)
    userEvent.click(screen.getByRole('button'))
    expect(mockedFn).toHaveBeenCalledTimes(1)
  })
})
