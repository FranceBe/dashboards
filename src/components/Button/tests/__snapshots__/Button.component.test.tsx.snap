// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot when disabled 1`] = `
<button
  class="sc-bdnxRM dcFAlz"
  disabled=""
>
  Button text
</button>
`;

exports[`Button should match snapshot when enabled 1`] = `
<button
  class="sc-bdnxRM dcFAlz"
>
  Button text
</button>
`;
